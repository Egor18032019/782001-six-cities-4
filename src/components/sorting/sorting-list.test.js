import React from "react";
import renderer from "react-test-renderer";
import SortingList from "./sorting-list.jsx";


describe(`SortingList snepshot test`, () => {
  it(`Should SortingList render correctly and will be open`, () => {
    const tree = renderer
      .create(<SortingList
        isOpen = {
          true
        }
        typeSorting = {
          `Popular`
        }
        onSelectClick = {
          () => {}
        }
        onSelectItemClick = {
          () => {}
        }
        handleClickOutside = {
          () => {}
        }
      />,
      // так как нет контейнера куда отрисовываться = делаем мокковый
      {
        createNodeMock: () => document.createElement(`form`)
      })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
  it(`Should SortingList render correctly and will be close`, () => {
    const tree = renderer
      .create(<SortingList
        isOpen = {
          false
        }
        typeSorting = {
          `Popular`
        }
        onSelectClick = {
          () => {}
        }
        onSelectItemClick = {
          () => {}
        }
        handleClickOutside = {
          () => {}
        }
      />,
      // так как нет контейнера куда отрисовываться = делаем мокковый
      {
        createNodeMock: () => document.createElement(`form`)
      })
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
