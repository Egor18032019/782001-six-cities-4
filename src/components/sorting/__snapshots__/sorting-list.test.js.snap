// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortingList snepshot test Should SortingList render correctly and will be close 1`] = `
<ul
  className="places__options places__options--custom "
>
  <li
    className="places__option places__option--active"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
  </li>
  <li
    className="places__option "
    onClick={[Function]}
    tabIndex="0"
  >
    Price: low to high
  </li>
  <li
    className="places__option "
    onClick={[Function]}
    tabIndex="0"
  >
    Price: high to low
  </li>
  <li
    className="places__option "
    onClick={[Function]}
    tabIndex="0"
  >
    Top rated first
  </li>
</ul>
`;

exports[`SortingList snepshot test Should SortingList render correctly and will be open 1`] = `
<ul
  className="places__options places__options--custom places__options--opened"
>
  <li
    className="places__option places__option--active"
    onClick={[Function]}
    tabIndex="0"
  >
    Popular
  </li>
  <li
    className="places__option "
    onClick={[Function]}
    tabIndex="0"
  >
    Price: low to high
  </li>
  <li
    className="places__option "
    onClick={[Function]}
    tabIndex="0"
  >
    Price: high to low
  </li>
  <li
    className="places__option "
    onClick={[Function]}
    tabIndex="0"
  >
    Top rated first
  </li>
</ul>
`;
